Week7/CurrentNews/app/src/main/java/com/florencenjeri/cocktailsrecipe/database/NewsRepository.kt

```diff
@@ -1,18 +1,28 @@
 package com.florencenjeri.cocktailsrecipe.database
 
-import androidx.lifecycle.LiveData
+import android.net.ConnectivityManager
 import com.florencenjeri.cocktailsrecipe.App
-import com.florencenjeri.cocktailsrecipe.model.New
 import com.florencenjeri.cocktailsrecipe.model.Success
+import com.florencenjeri.cocktailsrecipe.network.NetworkStatusChecker
 
 class NewsRepository() {
-    fun fetchNews(): LiveData<List<New>> = App.newsDao.fetchNews()
+    companion object {
+        private const val PAGE_SIZE = 30
+    }
+
+    private val networkStatusChecker by lazy {
+        NetworkStatusChecker(App.getAppContext().getSystemService(ConnectivityManager::class.java))
+    }
+
+    fun fetchNews() = App.newsDao.fetchNews()
 
     suspend fun insertNews() {
```

Comment:
I would rename this, as it says insertNews, but doesn't take in any parameters. 

Maybe refreshNews or something would be better!